contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+                    // non zero amount attempts to create a new instance of the pool
+                    sqrtPriceX96: 0
+                })
+            );
+
+            tokenId = mintedTokenId;
+
+            updateUnderlyingAssets();
+            emit LiquidityAdded(
+                WETHBalance, // WETHAmountDesired
+                OETHbRequired, // OETHbAmountDesired
+                WETHAmountSupplied, // WETHAmountSupplied
+                OETHbAmountSupplied, // OETHbAmountSupplied
+                mintedTokenId, // tokenId
+                underlyingAssets
+            );
+        } else {
+
+            (,uint256 WETHAmountSupplied, uint256 OETHbAmountSupplied) = positionManager.increaseLiquidity(
+                INonfungiblePositionManager.IncreaseLiquidityParams({
+                    tokenId: tokenId,
+                    amount0Desired: WETHBalance,
+                    amount1Desired: OETHbRequired,
+                    // we don't need slippage protection checking for that in _checkLiquidityWithinExpectedShare
+                    amount0Min: 0,
+                    // we don't need slippage protection checking for that in _checkLiquidityWithinExpectedShare
+                    amount1Min: 0,
+                    deadline: block.timestamp
+                })
+            );
+
+            updateUnderlyingAssets();
+            emit LiquidityAdded(
+                WETHBalance, // WETHAmountDesired
+                OETHbRequired, // OETHbAmountDesired
+                WETHAmountSupplied, // WETHAmountSupplied
+                OETHbAmountSupplied, // OETHbAmountSupplied
+                tokenId, // tokenId
+                underlyingAssets
+            );
+        }
+
+        // burn remaining OETHb
+        _burnOETHbOnTheContract();
+        positionManager.approve(address(clGauge), tokenId);
+        clGauge.deposit(tokenId);
+    }
+
+    /**
+     * @dev Check that the liquidity in the pool is withing the expected WETH to OETHb ratio
+     */
+    function _checkLiquidityWithinExpectedShare() internal {
+        (uint256 WETHPositionBalance, uint256 OETHbPositionBalance) = getPositionPrincipal();
+        require(WETHPositionBalance + OETHbPositionBalance > 0, "Can not withdraw full position");
+
+        uint160 currentPrice = getPoolX96Price();
+        // check we are in inspected tick range
+        require(currentPrice >= sqrtRatioX96Tick0 && currentPrice <= sqrtRatioX96Tick1, "Not in expected tick range");
+
+        uint256 currentWethShare = 0;
+        if (WETHPositionBalance != 0) {
+            currentWethShare = WETHPositionBalance.divPrecisely(WETHPositionBalance + OETHbPositionBalance);
+        }
+
+        if (currentWethShare < poolWethShareVarianceAllowed || // uint256 mustn't go below 0  
+            poolWethShare < currentWethShare - poolWethShareVarianceAllowed ||
+            poolWethShare > currentWethShare + poolWethShareVarianceAllowed) {
+
+            revert PoolRebalanceOutOfBounds(
```

Comment:
nit: Thought we are avoiding custom errors in this repo

Reply:
Yeah I know, though I still want to use them where the revert should convey important variables that will help us debug the source of the problem.